AmplifyPlugins/API/AWSAPICategoryPluginFunctionalTests/GraphQLModelBased/GraphQLConnectionScenario3Tests.swift

```diff
@@ -152,6 +153,83 @@ class GraphQLConnectionScenario3Tests: XCTestCase {
         wait(for: [updateCommentSuccessful], timeout: TestCommonConstants.networkTimeout)
     }
 
+    func testUpdateExistingPost() {
+        let uuid = UUID().uuidString
+        let testMethodName = String("\(#function)".dropLast(2))
+        let title = testMethodName + "Title"
+        guard var post = createPost(id: uuid, title: title) else {
+            XCTFail("Failed to ensure at least one Post to be retrieved on the listQuery")
+            return
+        }
+        let updatedTitle = title + "Updated"
+        post.title = updatedTitle
+        let completeInvoked = expectation(description: "request completed")
+        _ = Amplify.API.mutate(request: .update(post)) { event in
+            switch event {
+            case .success(let data):
+                switch data {
+                case .success(let post):
+                    XCTAssertEqual(post.title, updatedTitle)
+                case .failure(let error):
+                    print(error)
+                }
+                completeInvoked.fulfill()
+            case .failure(let error):
+                print(error)
+            }
+        }
+        wait(for: [completeInvoked], timeout: TestCommonConstants.networkTimeout)
+    }
+
+    func testDeleteExistingPost() {
+        let uuid = UUID().uuidString
+        let testMethodName = String("\(#function)".dropLast(2))
+        let title = testMethodName + "Title"
+        guard let post = createPost(id: uuid, title: title) else {
+            XCTFail("Failed to ensure at least one Post to be retrieved on the listQuery")
+            return
+        }
+
+        let completeInvoked = expectation(description: "request completed")
+
+        _ = Amplify.API.mutate(request: .delete(post)) { event in
+            switch event {
+            case .success(let data):
+                switch data {
+                case .success(let post):
+                    XCTAssertEqual(post.title, title)
+                case .failure(let error):
+                    print(error)
+                }
+                completeInvoked.fulfill()
+            case .failure(let error):
+                print(error)
+            }
+        }
+        wait(for: [completeInvoked], timeout: TestCommonConstants.networkTimeout)
+
```

Comment:
If we fail to wait for completeInvoked, i think the test will continue to execute here (right?)... is this what we want, or should we just fail?

Reply:
if it fails to wait for the expectation, it should fail the test and shouldn't continue to execute. we want it to continue when the expectation is met to then query for the deleted post to check that it's not there